packages/x-tree-view/src/useTreeItem2/useTreeItem2.types.ts

```diff
@@ -56,6 +57,7 @@ export type UseTreeItem2RootSlotProps<ExternalProps = {}> = ExternalProps &
 
 export interface UseTreeItem2ContentSlotOwnProps {
   onClick: MuiCancellableEventHandler<React.MouseEvent>;
+  onDoubleClick: MuiCancellableEventHandler<React.MouseEvent>;
```

Comment:
Would it make sense to have the trigger on the label slot instead?
So that we don't trigger the editing if people have some custom interactive UI outside of the label?

Reply:
I think it does make sense. I made the switch 🤔